Misc/NEWS.d/next/Library/2020-05-23-12-05-15.bpo-38632.nFTEqW.rst

```diff
@@ -0,0 +1,5 @@
+Partial support for ``SOURCE_DATE_EPOCH`` environment variable for sdists
+has been added in distutils. When the ``SOURCE_DATE_EPOCH`` environment
+variable is set, the ``mtime`` of the files in an sdist tar archive will not
+be later than ``SOURCE_DATE_EPOCH``. This is a firs step to simplify getting
+byte identical reproducibility of source dists.
```

Comment:
1) Can you explain why this PR is `a first step` / `partial support`?

2) distutils is mostly invisible these days, importated and extended by setuptools.  Do the changes impact setuptools sdists?

Reply:
There a bunch of other things that add variability to the end artifact. For example ZIP using current datetime in the header, which also needs to be fixed, making sure directory/files are traveres in the same order... etc.

I try to keep things minimal as PRs here tends to sits for years, and the bigger the longer. 


> 2\. distutils is mostly invisible these days, imported and extended by setuptools.  Do the changes impact setuptools sdists?

Yes, it _extends_ and call it, so if the result of this is not reproducible then it needs to be monkeypatch in setuptools (see https://github.com/pypa/setuptools/pull/2136 which I need to finish) Basically 1/2 of the upstream patch is copy-pasting code from distutils to tweak the internal behavior.

Reply:
That sounds OK then!